### PR TITLE
Star rating

### PR DESCRIPTION
**Pull request by saifudinbadani**
Please review rating component.

Screen Shot 2022-03-29 at 5 50 07 pm

### CODE REVIEW

**Comment by pritam-kr** [comment]
Good to Merge.

**Review by pritam-kr — APPROVED** [review]

File: components/rating/rating.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const ratingStar = [...document.getElementsByClassName('rating-star')];
+
+// reference
+// https://dev.to/leonardoschmittk/how-to-make-a-star-rating-with-js-36d3
+ratingStar.map((item) => {
+    starActive = 'rating-star rating-star-checked fas fa-star';
+    starInactive = 'rating-star far fa-star';
+
+    item.addEventListener('click', () => {
+        let i = ratingStar.indexOf(item);
```
Comment (sahiba0915) [inline components/rating/rating.js:10]: can use const though

**Review by sahiba0915 — APPROVED** [review]
Looks good to me given one suggestion, liked the code structure, code for dynamic ratings and used good naming conventions

File: components/rating/rating.html [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            <code class="language-markup">
+                <div class="rating-star-container">
+                    <i class="rating-star far fa-star"></i>
+                    <i class="rating-star far fa-star"></i>
+                    <i class="rating-star far fa-star"></i>
+                    <i class="rating-star far fa-star"></i>
+                    <i class="rating-star far fa-star"></i>
+                </div>
+
+                <!-- code used for changing rating dynamically: -->
```
Comment (utkarshducoder) [inline components/rating/rating.html:75]: try to remove these comments as it takes extra space

**Review by utkarshducoder — APPROVED** [review]

File: components/rating/rating.html [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                // reference
```
Comment (palash23jain) [inline components/rating/rating.html:76]: good to see you adding the link of reference as a comment. Good practice

**Review by palash23jain — APPROVED** [review]
Looks good to me

File: components/rating/rating.html (hunk 2 above)
Comment (saifudinbadani) [reply components/rating/rating.html:75]: It is written so the user knows what does below the code do..

File: components/rating/rating.js (hunk 1 above)
Comment (saifudinbadani) [reply components/rating/rating.js:10]: the value of variable is changing that is why i used let.

**Pull request merged**